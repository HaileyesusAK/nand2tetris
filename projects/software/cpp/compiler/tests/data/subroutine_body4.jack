{
	var boolean a;
	let a = false;
	while(~a) {
		let a = true;
	}
}
