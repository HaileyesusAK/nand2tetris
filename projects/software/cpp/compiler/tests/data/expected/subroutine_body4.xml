<subroutineBody>
    <symbol> { </symbol>
    <varDec>
        <keyword> var </keyword>
        <keyword> boolean </keyword>
        <identifier> a </identifier>
        <symbol> ; </symbol>
    </varDec>
    <statements>
        <letStatment>
            <keyword> let </keyword>
            <identifier> a </identifier>
            <symbol> = </symbol>
            <expression>
                <term>
                    <keyword> false </keyword>
                </term>
            </expression>
            <symbol> ; </symbol>
        </letStatement>
        <whileStatement>
            <keyword> while </keyword>
            <symbol> ( </symbol>
            <expression>
                <term>
                    <symbol> ~ </symbol>
                    <term>
                        <identifier> a </identifier>
                    </term>
                </term>
            </expression>
            <symbol> ) </symbol>
            <symbol> { </symbol>
            <statements>
                <letStatment>
                    <keyword> let </keyword>
                    <identifier> a </identifier>
                    <symbol> = </symbol>
                    <expression>
                        <term>
                            <keyword> true </keyword>
                        </term>
                    </expression>
                    <symbol> ; </symbol>
                </letStatement>
            </statements>
            <symbol> } </symbol>
        </whileStatement>
    </statements>
    <symbol> } </symbol>
</subroutineBody>
